aleph/views/roles_api.py

```diff
@@ -59,28 +60,31 @@ def create():
     require(not request.authz.in_maintenance,
             get_config('PASSWORD_REGISTRATION'))
     data = parse_request(schema=RoleCreateSchema)
+
     try:
-        code = data.get('code')
-        email = Role.SIGNATURE.loads(code, max_age=Role.SIGNATURE_MAX_AGE)
-        assert email == data.get('email')
-    except Exception:
-        raise BadRequest("Invalid signature")
-
-    role = Role.by_email(email).first()
-    status = 200
-    if role is None:
-        status = 201
-        role = Role.load_or_create(
-            foreign_id='password:{}'.format(email),
-            type=Role.USER,
-            name=data.get('name') or email,
-            email=email
-        )
-        role.set_password(data.get('password'))
-        db.session.add(role)
-        db.session.commit()
-    request.authz.id = role.id
-    return jsonify(role, schema=RoleSchema, status=status)
+        email = Role.SIGNATURE.loads(data.get('code'), max_age=Role.SIGNATURE_MAX_AGE)
+
+        role = Role.by_email(email).first()
+        if role is None:
+            role = Role.load_or_create(
+                foreign_id='password:{}'.format(email),
+                type=Role.USER,
+                name=data.get('name') or email,
+                email=email
+            )
+            role.set_password(data.get('password'))
+            db.session.add(role)
+            db.session.commit()
+            # Let the serializer return more info about this user
+            request.authz.id = role.id
```

Comment:
This feels like magic, should it be here? We need to return the email to the user and currently there is some logic [here](https://github.com/alephdata/aleph/blob/master/aleph/views/serializers.py#L79) which removes the `email` if they aren't authed.

This makes sense in theory, but the `foreign_id` also contains the email address and is always serialised. Should it be removed too?